cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -198,14 +198,19 @@ void LocalTrajectoryBuilder::AddImuData(const sensor::ImuData& imu_data) {
   extrapolator_->AddImuData(imu_data);
 }
 
-void LocalTrajectoryBuilder::AddOdometerData(
+std::unique_ptr<sensor::OdometryData> LocalTrajectoryBuilder::AddOdometerData(
     const sensor::OdometryData& odometry_data) {
   if (extrapolator_ == nullptr) {
     // Until we've initialized the extrapolator we cannot add odometry data.
     LOG(INFO) << "Extrapolator not yet initialized.";
-    return;
+    return nullptr;
   }
+  const transform::Rigid3d gravity_alignment = transform::Rigid3d::Rotation(
+      extrapolator_->EstimateGravityOrientation(odometry_data.time));
```

Comment:
Should this be called before or after adding the odometry data to the extrapolator?